api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListOfPatientsCurrentlyOnArtWithoutTbScreeningCohortQueries.java

```diff
@@ -0,0 +1,580 @@
+   */
+  public DataDefinition getNextScheduledDrugPickupDateOnRecepcaoLevantouArv() {
+
+    SqlPatientDataDefinition sqlPatientDataDefinition = new SqlPatientDataDefinition();
+    sqlPatientDataDefinition.setName("Most Recent Drug Pick-Up on Recepção Levantou ARVs ");
+    sqlPatientDataDefinition.addParameter(new Parameter("location", "Location", Location.class));
+    sqlPatientDataDefinition.addParameter(new Parameter("endDate", "End Date", Location.class));
+
+    Map<String, Integer> valuesMap = new HashMap<>();
+    valuesMap.put("23866", hivMetadata.getArtDatePickupMasterCard().getConceptId());
+    valuesMap.put("52", hivMetadata.getMasterCardDrugPickupEncounterType().getEncounterTypeId());
+
+    String query =
+        "SELECT p.patient_id, "
+            + "       MAX(DATE_ADD(o.value_datetime, INTERVAL 30 MONTH)) encounter_date "
+            + "FROM   patient p "
+            + "       INNER JOIN encounter e "
+            + "               ON e.patient_id = p.patient_id "
+            + "       INNER JOIN obs o "
+            + "               ON o.encounter_id = e.encounter_id "
+            + "WHERE  e.encounter_type = ${52} "
+            + "       AND e.location_id = :location "
+            + "       AND e.encounter_datetime <= :endDate "
+            + "       AND o.concept_id = ${23866} "
+            + "       AND p.voided = 0 "
+            + "       AND e.voided = 0 "
+            + "GROUP  BY p.patient_id";
+
+    StringSubstitutor sb = new StringSubstitutor(valuesMap);
+    sqlPatientDataDefinition.setQuery(sb.replace(query));
+
+    return sqlPatientDataDefinition;
+  }
+
+  /**
+   * <b>Technical Specs</b>
+   *
+   * <blockquote>
+   *
+   * <p>
+   *
+   * <p>Patient’s Most Recent Ficha with MDCs Registered
+   * <p>Encounter DATE (encounter.encounter_datetime)</p>
+   * <p>Modo De Dispensa (id= 165174)</p>
+   *
+   * </blockquote>
+   *
+   * @return {@link DataDefinition}
+   */
+  public DataDefinition getMostRecentMdcConsultationDate() {
+
+    SqlPatientDataDefinition sqlPatientDataDefinition = new SqlPatientDataDefinition();
+    sqlPatientDataDefinition.setName("MDC Consultation Date ");
+    sqlPatientDataDefinition.addParameter(new Parameter("location", "Location", Location.class));
+
+    Map<String, Integer> valuesMap = new HashMap<>();
+    valuesMap.put("6", hivMetadata.getAdultoSeguimentoEncounterType().getEncounterTypeId());
+    valuesMap.put("165174", hivMetadata.getLastRecordOfDispensingModeConcept().getConceptId());
+
+    String query =
+        "SELECT p.patient_id, Max(e.encounter_datetime) consultation_date "
+            + "FROM   patient p "
+            + "       INNER JOIN encounter e ON e.patient_id = p.patient_id "
+            + "       INNER JOIN obs o ON o.encounter_id = e.encounter_id "
+            + "WHERE  e.encounter_type = ${6} "
+            + "       AND e.location_id = :location "
+            + "       AND e.encounter_datetime <= CURRENT_DATE() "
```

Comment:
Do you want to compare against current time and date of you will need to have end of reporting period?

Reply:
Maybe report generation date could mean the same thing, current date when the report is being generated

Reply:
Implemented cording to the requirements

The system will identify patient’s most recent records of MDCs (DSD) registered in Ficha Clínica by report generation date and will show the registered MDC1, MDC2, MDC3, MDC4 and MDC5 if marked as “Inicio” or “Continua” as follows: